internal/server/iigointernal/orchestration.go

```diff
@@ -68,48 +68,48 @@ var presidentPointer roles.President = nil
 // iigoClients holds pointers to all the clients
 var iigoClients map[shared.ClientID]baseclient.Client
 
-var iigoRoleStates gamestate.IIGOBaseRoles
-
 // RunIIGO runs all iigo function in sequence
 func RunIIGO(g *gamestate.GameState, clientMap *map[shared.ClientID]baseclient.Client) error {
 
 	// TODO: Get Client pointers from gamestate https://imgur.com/a/HjVZIkh
 	iigoClients = *clientMap
-	iigoRoleStates = g.IIGOInfo
 
 	// Initialise IDs
-	featureJudge.ID = JudgeIDGlobal
-	featureSpeaker.Id = SpeakerIDGlobal
-	featurePresident.ID = PresidentIDGlobal
+	judicialBranch.JudgeID = JudgeIDGlobal
+	legislativeBranch.SpeakerID = SpeakerIDGlobal
+	executiveBranch.ID = PresidentIDGlobal
 
 	// Initialise iigointernal with their clientVersions
-	featureJudge.clientJudge = judgePointer
-	featurePresident.clientPresident = presidentPointer
-	featureSpeaker.clientSpeaker = speakerPointer
+	judicialBranch.clientJudge = judgePointer
+	executiveBranch.clientPresident = presidentPointer
+	legislativeBranch.clientSpeaker = speakerPointer
 
 	// Withdraw the salaries
-	errWithdrawPresident := featureJudge.withdrawPresidentSalary(g)
-	errWithdrawJudge := featureSpeaker.withdrawJudgeSalary(g)
-	errWithdrawSpeaker := featurePresident.withdrawSpeakerSalary(g)
+	errWithdrawPresident := judicialBranch.withdrawPresidentSalary(g)
+	errWithdrawJudge := legislativeBranch.withdrawJudgeSalary(g)
+	errWithdrawSpeaker := executiveBranch.withdrawSpeakerSalary(g)
 
 	// Handle the lack of resources
 	if errWithdrawPresident != nil {
+		returnWithdrawnSalariesToCommonPool(g)
 		return errors.Errorf("Could not run IIGO since President has no resoruces to spend")
 	}
 	if errWithdrawJudge != nil {
+		returnWithdrawnSalariesToCommonPool(g)
 		return errors.Errorf("Could not run IIGO since Judge has no resoruces to spend")
 	}
 	if errWithdrawSpeaker != nil {
+		returnWithdrawnSalariesToCommonPool(g)
 		return errors.Errorf("Could not run IIGO since Speaker has no resoruces to spend")
 	}
 
 	// Pay salaries into budgets
-	featureJudge.sendPresidentSalary()
-	featureSpeaker.sendJudgeSalary()
-	featurePresident.sendSpeakerSalary()
+	judicialBranch.sendPresidentSalary()
+	legislativeBranch.sendJudgeSalary()
+	executiveBranch.sendSpeakerSalary()
 
 	// 1 Judge actions - inspect history
-	_, judgeInspectingHistoryError := featureJudge.InspectHistory()
+	_, judgeInspectingHistoryError := judicialBranch.inspectHistory()
```

Comment:
Another issue, what if the judge decides not to inspect the history. Are they forced to here?